src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs

```diff
@@ -49,7 +49,9 @@ internal static string GenerateKey(ViewLocationExpanderContext context)
         {
             var keyBuilder = new StringBuilder();
             var routeValues = context.ActionContext.RouteData.Values;
-            var controller = routeValues.GetValueOrDefault<string>(RazorViewEngine.ControllerKey);
+            var controller = RouteValueUtility.GetNormalizedRouteValue(
```

Comment:
Although the only issue is the keys are `internal` and maybe this method has some utility in user code - for instance in a custom view location expander.